bgpd/bgp_route.c

```diff
@@ -8545,6 +8545,78 @@ DEFPY(aggregate_addressv4, aggregate_addressv4_cmd,
 				 match_med != NULL, suppress_map);
 }
 
+void bgp_cleanup_aggregate_info(struct bgp *bgp)
+{
+	struct bgp_aggregate *aggregate;
+	afi_t afi;
+	safi_t safi;
+
+	FOREACH_AFI_SAFI (afi, safi) {
+		for (struct bgp_dest *dest =
+			     bgp_table_top(bgp->aggregate[afi][safi]);
+		     dest; dest = bgp_route_next(dest)) {
+			aggregate = bgp_dest_get_bgp_aggregate_info(dest);
+			if (aggregate == NULL)
+				continue;
+			bgp_dest_set_bgp_aggregate_info(dest, NULL);
+
+			if (aggregate->community)
+				community_free(&aggregate->community);
+
+			if (aggregate->community_hash) {
+				/* Delete all communities in the hash.
+				 */
+				hash_clean(aggregate->community_hash,
```

Comment:
Replace hash_clean + hash_free with hash_clean_and_free().

Reply:
I have updated the function similar to bgp_aggregate_unset()